observations/obs_converters/AIRS/convert_amsu_L1.f90

```diff
@@ -107,31 +122,81 @@ program convert_amsu_L1
 filecount = set_filename_list(l1_files, l1_file_list, "convert_amsu_L1")
 
 ! use the first file to initialize the module
-call initialize_amsua_netcdf(l1_files(1))
+call initialize_amsua_netcdf(l1_files(1),verbose)
 
 ! FIXME ... maybe ... are all the files guaranteed to have the
 ! same channels - I think so.
 call channel_list_to_indices(channel_list, use_channels)
 
-! for each input file
+! Figure out how many (more) observations there may be in the output.
+! It is impossible to predict how the lat/lon subsetting may affect this
+! number, but the thinning and channels can be taken into account.
+
+max_num = max_possible_obs(filecount, cross_track_thin, along_track_thin, use_channels)
+
+! either read existing obs_seq or create a new one
+
+inquire(file=outputfile, exist=file_exist)
+
+if ( file_exist .and. append_output ) then
+  call read_obs_seq(outputfile, 0, 0, max_num, big_sequence)
+  write(string1,*)'Appending to "'//trim(outputfile)//'"'
+  write(string2,*)'Initially has ',get_num_obs(big_sequence),' observations.'
+else
+  call init_obs_sequence(big_sequence, NUM_COPIES, NUM_QC, max_num)
+  do i = 1, NUM_COPIES
+    call set_copy_meta_data(big_sequence, i, 'observation')
+  end do
+  do i = 1, NUM_QC
+    call set_qc_meta_data(big_sequence, i, 'QC')
+  end do
+  write(string1,*)'Creating "'//trim(outputfile)//'" from scratch.'
+  write(string2,*)'Initially has ',get_num_obs(big_sequence),' observations.'
+endif
+
+if (verbose > 0) call error_handler(E_MSG,source,string1,text2=string2)
+
+! read from netCDF file into a derived type that holds all the information
+! convert derived type information to DART sequence for that file
+! combine that sequence to the output sequence
+! release the file sequence memory in preparation for creating it new next iteration
+
 do ifile = 1,filecount
 
-   ! read from HDF file into a derived type that holds all the information
    call read_amsua_bt_netCDF_granule(l1_files(ifile), granule)   
 
-   ! convert derived type information to DART sequence
-   call make_obs_sequence(seq, granule, lon1, lon2, lat1, lat2, &
+   call make_obs_sequence(small_sequence, granule, lon1, lon2, lat1, lat2, &
```

Comment:
if the create step is moved out of the loop and obs are added to the same sequence then you don't have to combine them as a separate step.  if there's a reason for having to do it this way there needs to be some comment before this loop saying why this choice was made.  it is going to have worse performance for large obs counts.

Reply:
There are about 240 files per day, and using two channels from each file results in 2700 obs per file, 648000 obs per day.
With gfortran -O2 optimization on my laptop, it took less than 30 seconds to convert all 240 files in a single run - and that was using the (obs_def_rttov_mod:MAXrttovkey) initial metadata length of 100,000.

I guess I just felt the code was pretty clear by adding the extra routine and leaving the existing working code as untouched as possible.

Reply:
i don't know that leaving the existing code untouched is justification for this, but i am convinced by the performance argument.   someone doing a long assimilation may be converting months (years) of these obs, but a run time of 30 seconds a day seems like it would be acceptable in a batch job.

Reply:
As another data point, I 'appended' the exact same 240 files to itself, which has to be a rather awful insertion pattern. The run-time (only did it once, so take it with a grain of salt) ... was 57 seconds.  Seemed to me that it took a bit of time to read the existing sequence, and write the 2X sequence ... as opposed to reading and inserting the new obs. for what its worth ...